lib/internal/process/modules.js

```diff
@@ -2,14 +2,28 @@
 
 const {
   setImportModuleDynamicallyCallback,
-  setInitializeImportMetaObjectCallback
+  setInitializeImportMetaObjectCallback,
 } = internalBinding('module_wrap');
 
+const getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor;
+const hasOwnProperty = Function.call.bind(Object.prototype.hasOwnProperty);
+const apply = Reflect.apply;
+
+const errors = require('internal/errors');
 const { getURLFromFilePath } = require('internal/url');
 const Loader = require('internal/loader/Loader');
 const path = require('path');
 const { URL } = require('url');
 
+// fires a getter or reads the value off a descriptor
```

Comment:
What’s the need to support getters? node could mandate that these things a;l be data properties.

Reply:
@ljharb is there precedent of something that fails when passed a getter but not a data property?

Reply:
can we at least use Reflect.get instead of this whole function

Reply:
Not that i know of in the language. node be as restrictive here as it wants; it could even reject loaders whose resolve functions don’t have the right .length

Reply:
@devsnek nope, because that would separate the check for property existing and the reading in a way that proxies could see. They could change their keys between that, generate a new property descriptor between that, etc. This function is also a bit gross because property descriptors inherit from `Object` and people can taint `value` / `get` and we don't want to use `Reflect.get` because it would grab those as well:

```mjs
Object.prototype.value = 123;
Object.getOwnPropertyDescriptor({get x(){}}, 'x').value
```

Reply:
Length isn’t unreliable; but you’re right that it would preclude using rest.

Reply:
@ljharb 

```mjs
[((g = 0) => g), g => g].map(f => f.length)
// 0, 1
```

Reply:
That's not unreliable :-) defaults aren't part of the length.

Reply:
Confusing to the point of not being reliable. Well defined, just not reliable to my eyes.

```mjs
[((g = 0, h) => g)].map(f => f.length)
// 0
```

Reply:
That’s invalid iirc; defaults can’t come before non-defaults.

Reply:
Naw, it's fine:
![image](https://user-images.githubusercontent.com/2932786/36557025-f3a824e6-17bb-11e8-8052-c839d825c317.png)

Reply:
huh, TIL. Fair enough.